roxctl/declarativeconfig/lint/lint_test.go

```diff
@@ -0,0 +1,183 @@
+package lint
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"testing"
+
+	"github.com/stackrox/rox/pkg/errox"
+	"github.com/stackrox/rox/roxctl/common/mocks"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestLintCommand_Failure(t *testing.T) {
+	cases := map[string]struct {
+		yaml []byte
+		err  error
+	}{
+		"invalid YAML": {
+			yaml: []byte(`
+X
+`),
+		},
+		"unknown configuration type": {
+			yaml: []byte(`name: test-name
+description: test-description
+policy:
+  name: my-policy
+  enforcement: DEPLOY_TIME`),
+		},
+		"invalid configuration during unmarshal": {
+			yaml: []byte(`name: test-name
+description: test-description
+resources:
+- resource: a
+  access: INVALID_ACCESS
+`),
+			err: errox.InvalidArgs,
+		},
+		"invalid configuration during transformation": {
+			yaml: []byte(`name: test-name
+minimumRole: "None"
+uiEndpoint: "localhost:8000"
+`),
+			err: errox.InvalidArgs,
+		},
+	}
+
+	for name, c := range cases {
+		t.Run(name, func(t *testing.T) {
+			l := lintCmd{fileContents: c.yaml}
+			err := l.Lint()
+			assert.Error(t, err)
+			if c.err != nil {
+				assert.ErrorIs(t, err, c.err)
+			}
+		})
+	}
+}
+
+func TestLintCommand_Success(t *testing.T) {
+	cases := map[string]struct {
+		yaml []byte
+	}{
+		"single configuration in file": {
+			yaml: []byte(`name: some-name
+minimumRole: Analyst
+uiEndpoint: localhost:8000
+extraUIEndpoints:
+    - localhost:9090
+groups:
+    - key: email
+      value: example@example.com
+      role: Admin
+requiredAttributes:
+    - key: org_id
+      value: "12345"
+    - key: name
+      value: some_name
+claimMappings:
+    - path: org_id
+      name: super_cool_claim
+    - path: republic
+      name: far_away
+oidc:
+    issuer: sample.issuer.com
+    mode: auto
+    clientID: CLIENT_ID
+    clientSecret: CLIENT_SECRET
+`),
+		},
+		"multiple configurations in file": {
+			yaml: []byte(`- name: some-name
```

Comment:
Looking at this example just now, I wonder if different declarative resource types would be more distinguishable if we forced to add `type: <resource type>` to every declarative resource. Example:
```
- name: ps
  type: PermissionSet:
  resources:
	...
- name: RHSSO
  type: AuthProvider
  ....
```

Reply:
Well, I suppose we could have done this (similar to how we have [ROX-14652](https://issues.redhat.com/browse/ROX-14652) for adding the metadata field, this could include the type field).

However, I think it's not worthwhile to add it anymore. Since post-MVP it's likely we move to CRs for the definition of declarative configuation, they would have the type inferred by their `Kind`.